thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/formatter/DetectionConfigFormatter.java

```diff
@@ -154,6 +170,20 @@ public DetectionConfigFormatter(MetricConfigManager metricDAO, DatasetConfigMana
     return metricUrns;
   }
 
+  private DetectionHealth getDetectionHealth(DetectionConfigDTO config) {
+    // Return the stored detection health when it is available in the DetectionConfig.
+    if (!Objects.isNull(config.getHealth())) {
+      return config.getHealth();
+    }
+    DateTime end = DateTime.now();
```

Comment:
No need to calculate here.
We can wait for the background task to update the status.

Reply:
Sure. I've changed it to return an Unknown status if it's not generated yet.